### PR TITLE
Post about 0.9 and family reunion

### PR DESCRIPTION
**Pull request by cwfitzgerald**
This is very incomplete, so feel free to add more by pushing to my branch.

### CODE REVIEW

File: _posts/2021-06-19-release-0.9.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+title: Release of v0.9
+---
+
+gfx-rs community's goal is to make graphics programming in Rust easy, fast, and reliable. Our current main projects are:
+
+  - [gfx-rs](https://github.com/gfx-rs/gfx) makes low-level GPU programming portable with low overhead. It's a single Vulkan-like Rust API with multiple backends that implement it: Direct3D 12/11, Metal, Vulkan, and even OpenGL ES.
+  - [naga](https://github.com/gfx-rs/naga) translates the shaders between languages, including WGSL. Also provides validation and processing facilities on the intermediate representation.
+  - [wgpu](https://github.com/gfx-rs/wgpu) is built on top of gfx-rs and [gpu-alloc](https://github.com/zakarumych/gpu-alloc)/[gpu-descriptor](https://github.com/zakarumych/gpu-descriptor). It provides safety, accessibility, and strong portability of applications.
```
Comment (kvark) [inline _posts/2021-06-19-release-0.9.md:10]: note: for the future entries, we should probably be less vocal about gpu-alloc/gpu-descriptor, since they are just backend details now with wgpu-hal.
But for this release it's definitely spelled right!

File: _posts/2021-06-19-release-0.9.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Following the regular schedule of releasing every few months, we just rolled out the 0.9 version through all of the gfx projects!
+See [gfx-rs changelist](https://github.com/gfx-rs/gfx/blob/master/CHANGELOG.md#hal-090-18-06-2021), [wgpu changelist](https://github.com/gfx-rs/wgpu/blob/master/CHANGELOG.md#v09-2021-06-18), and [naga changelist](https://github.com/gfx-rs/naga/blob/master/CHANGELOG.md#v05-2021-06-18) for the details.
+
+Naga has matured signifgantly since the last release. <naga>
+
+`wgpu` validation has continued continued to improve. Many validation holes were plugged with the last release. Through the combined work in `wgpu` and `naga`, validation holes have been sured up.
```
Comment (kvark) [inline _posts/2021-06-19-release-0.9.md:17]: "continued" twice

File: _posts/2021-06-19-release-0.9.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+`wgpu` perforamnce is still a vital target for us, so we have done work on improving the overhead at resource. <splitting tracker>
```
Comment (kvark) [inline _posts/2021-06-19-release-0.9.md:19]: "perforamnce" typo

Comment (kvark) [inline _posts/2021-06-19-release-0.9.md:19]: Might mention that it was Gecko profiling on WebGPU that drove the changes. People always appreciate when there are real use-cases involved, and semi-production use of the library.

File: _posts/2021-06-19-release-0.9.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## wgpu family reunion and relicense
+
+The biggest internal change since the last release was the progress towards the gfx-rs family reunion. Contributors have long struggled with the amount of effort it takes to add features to wgpu. They would have to make a PR against gfx-hal, then wgpu, then wgpu-rs.
```
Comment (kvark) [inline _posts/2021-06-19-release-0.9.md:23]: here is a question: do you intend to describe changes for 0.9 specifically? this is only moving wgpu-rs in.
Or do you intend to explain the whole set of changes here? In this case, we should make a bigger deal of it than just "biggest change since the last release" :)

**Review by kvark — COMMENTED** [review]
Thank you for taking a stab at this!

File: _posts/2021-07-16-release-0.9-future.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  - [naga](https://github.com/gfx-rs/naga) translates the shaders between languages, including WGSL. Also provides validation and processing facilities on the intermediate representation.
+  - [wgpu](https://github.com/gfx-rs/wgpu) is built on top of gfx-rs and [gpu-alloc](https://github.com/zakarumych/gpu-alloc)/[gpu-descriptor](https://github.com/zakarumych/gpu-descriptor). It provides safety, accessibility, and strong portability of applications.
+
+Following our release cadence every few months, we rolled out the 0.9 version through all of the gfx projects!
+See [gfx-rs changelog](https://github.com/gfx-rs/gfx/blob/master/CHANGELOG.md#hal-090-18-06-2021), [wgpu changelog](https://github.com/gfx-rs/wgpu/blob/master/CHANGELOG.md#v09-2021-06-18), and [naga changelog](https://github.com/gfx-rs/naga/blob/master/CHANGELOG.md#v05-2021-06-18) for the details.
+
+`naga` has matured signifgantly since the last release. `wgsl` parsing has imporved incredibly, targetting an up-to-date spec. `spirv` parsing support has had numerous bugs fixed, and `glsl` support is starting to take shape, though still in an alpha state.
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:15]: "signifgantly"

Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:15]: should this become a list?
>`naga` has matured signifgantly since the last release:
>  - `wgsl` ...
>  - `spirv` ..., but still has major open issues with control flow
>  - `glsl` ...
>  - general validation and correctness in other endpoints

File: _posts/2021-07-16-release-0.9-future.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+`wgpu` validation has continued to improve. Many validation holes were plugged with the last release. Through the combined work in `wgpu` and `naga`, validation holes have been sured up.
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:17]: > An example of such collaboration is support for getting the array length of runtime-sized arrays, which is now implemented on Metal.

Maybe not exactly the validation thing, but could be nice to mention?

File: _posts/2021-07-16-release-0.9-future.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+`wgpu` performance is still a vital target for us, so we have done work on improving the overhead at resource tracking. We've reduced unnecisary overhead through only doing stateful tracking for resources that have complex states. These changes were made from benchmarks of Gecko's WebGPU implemention which showed that tracking was a bottleneck. You can read more about it [#1413](https://github.com/gfx-rs/wgpu/issues/1413).
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:19]: "overhead at resource tracking" -> maybe "overhead of resource tracking"?

File: _posts/2021-07-16-release-0.9-future.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## Thank You!
+
+Thank you for the countless contributors that helped out with this release! `wgpu`'s momentum is only increasing due to everyone's contirbutions and we look forward to seeing the amazing places wgpu will go as a project. If you are interested in helping, take a look at our [good-first-issues](https://github.com/gfx-rs/wgpu/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22), our issues with [help wanted](https://github.com/gfx-rs/wgpu/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22help+wanted%22) or contact us on our [matrix chat](https://matrix.to/#/#wgpu:matrix.org), we are always willing to help mentor first time and returning contributors.
+
+Additionally, thank you to all the users who report new issues, ask for enhancements, or test the git version of wgpu. Keep it coming!
+
+Happy rendering!
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:39]: going to add this to my vocabulary now :)

File: _posts/2021-07-16-release-0.9-future.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+Finally, we have brand new testing infrastructure that allows us to automatically test across all backends and all adapters in the system. Included in our tests are image comparison tests for all of our examples and the beginnings of feature tests. We hope to expand this to cover a wide variety of features and use cases. We will be able to run these tests in CI on software adapters and our future goal is to setting up a distributed testing network so that we can automatically test on a wide range of adapters. This will be one important layer of our in-depth defences, ensuring that wgpu is actually portable and safe. Numerous bugs have already been caught by this new infrastrucutre thus far and it will help us prevent regressions in the future. [Read more about our testing infrastructure](https://github.com/gfx-rs/wgpu/discussions/1611)
+
+## Thank You!
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:33]: nice to have this section!
would it make sense to align the capitalization between sections? This one is all capital, but the other one is all lowercase.

File: _posts/2021-07-16-release-0.9-future.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+`wgpu` validation has continued to improve. Many validation holes were plugged with the last release. Through the combined work in `wgpu` and `naga`, validation holes have been sured up.
+
+`wgpu` performance is still a vital target for us, so we have done work on improving the overhead at resource tracking. We've reduced unnecisary overhead through only doing stateful tracking for resources that have complex states. These changes were made from benchmarks of Gecko's WebGPU implemention which showed that tracking was a bottleneck. You can read more about it [#1413](https://github.com/gfx-rs/wgpu/issues/1413).
+
+## wgpu family reunion, relicense, and the future
+
+`wgpu` has had a number of large internal changes which are laying the future for wgpu to be a safe, efficient, and portable api for doing cross-platform graphics.
+
+`wgpu` has been relicensed from `MPL-2.0` to `MIT/Apache-2.0`. Thank you to all 142 people who replied to the issue and made this happen. This relicense is an important change because it allows the posibility of adding backends that target APIs that are behind NDAs.
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:25]: "adding backends that target APIs that are" - maybe reword this to avoid repeating "that"

File: _posts/2021-07-16-release-0.9-future.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+Thanks to feedback from our friends at the bevy game engine, we launched a initiative to make wgpu easier to contribute to. We moved wgpu-rs back into the `wgpu` repo. This means that PRs that touch both the core crate and the rust bindings no longer need multiple PRs that need to be synchronized. We have already heard from collaborators how much easier the contribution is now that there is less coordination to do. [Read more about the family reunion](https://github.com/gfx-rs/wgpu/milestone/9)
```
Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:27]: maybe capitalize `Bevy` and/or make it a link?

Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:27]: I wonder if this needs to be reworded to put less stress on the Bevy discussion. What happened really was that it has given us the impulse to address the problems we've already been aware of for a long time. It was the "last straw", even if it was a relatively big straw :)

> For a while, we acknowledged that having different essential parts of the project living in different repositories was hurting developers productivity. There were objective reasons for this, but the time has come to change that. Finally, thanks to feedback from our friends at the bevy game engine, we launched an initiative to make wgpu easier to contribute to.

Does that sound right?

Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:27]: "a initiative" -> "an initiative"

Comment (kvark) [inline _posts/2021-07-16-release-0.9-future.md:27]: the family reunion link - let's point to the closed issues list https://github.com/gfx-rs/wgpu/milestone/9?closed=1

**Review by kvark — COMMENTED** [review]
Very nice!

File: _posts/2021-07-16-release-0.9-future.md (hunk 11 above)
Comment (cwfitzgerald) [reply _posts/2021-07-16-release-0.9-future.md:27]: Yeah, that sounds about right

File: _posts/2021-07-16-release-0.9-future.md (hunk 10 above)
Comment (cwfitzgerald) [reply _posts/2021-07-16-release-0.9-future.md:25]: I'll do you one better and get rid of them all ;)

**Review by kvark — APPROVED** [review]
